pkg/server/api/entry/v1/service.go

```diff
@@ -169,6 +175,83 @@ func (s *Service) BatchDeleteEntry(ctx context.Context, req *entry.BatchDeleteEn
 	}, nil
 }
 
+func (s *Service) updateEntry(ctx context.Context, e *types.Entry, inputMask *types.EntryMask, outputMask *types.EntryMask) *entry.BatchUpdateEntryResponse_Result {
+	log := rpccontext.Logger(ctx)
+
+	if e == nil {
+		log.Error("missing entry")
+		return &entry.BatchUpdateEntryResponse_Result{
+			Status: api.CreateStatus(codes.Internal, "missing entry"),
+		}
+	}
+
+	if inputMask == nil {
+		log.Error("missing input mask")
+		return &entry.BatchUpdateEntryResponse_Result{
+			Status: api.CreateStatus(codes.Internal, "missing input mask"),
+		}
+	}
+	if e.SpiffeId != nil {
+		log = log.WithField(telemetry.SPIFFEID, e.SpiffeId)
+	}
+	// Fetch the previous version of the entry.
+	res, err := s.ds.FetchRegistrationEntry(ctx,
```

Comment:
not sure about this approach :(, maybe we can add a mask to ds and apply changes there? to avoid fetching each entries multiple times?

Reply:
That could work
Or if the mask is nil/all, we don't have to fetch? I think probably most people will mostly be updating all fields?

Reply:
I'm happy to push this into the datastore layer if we think that's the right place to put it though